components/datetime/src/datetime.rs

```diff
@@ -262,23 +180,21 @@ impl DateTimeFormat {
     /// use icu::locid::Locale;
     /// use icu::locid::macros::langid;
     /// use icu::datetime::{DateTimeFormat, DateTimeFormatOptions};
-    /// use icu::calendar::DateTime;
+    /// use icu::calendar::{DateTime, Gregorian};
     /// use icu_provider::inv::InvariantDataProvider;
     /// # let locale: Locale = langid!("en").into();
     /// # let provider = InvariantDataProvider;
     /// # let options = DateTimeFormatOptions::default();
-    /// let dtf = DateTimeFormat::try_new(locale, &provider, &options.into())
+    /// let dtf = DateTimeFormat::<Gregorian>::try_new(locale, &provider, &options.into())
     ///     .expect("Failed to create DateTimeFormat instance.");
     ///
     /// let datetime = DateTime::new_gregorian_datetime_from_integers(2020, 9, 1, 12, 34, 28)
     ///     .expect("Failed to construct DateTime.");
     ///
     /// let _ = dtf.format_to_string(&datetime);
     /// ```
-    pub fn format_to_string(&self, value: &impl DateTimeInput) -> String {
-        let mut s = String::new();
-        self.format_to_write(&mut s, value)
-            .expect("Failed to write to a String.");
-        s
+    #[inline]
+    pub fn format_to_string(&self, value: &impl DateTimeInput<Calendar = C>) -> String {
+        self.0.format_to_string(value)
     }
```

Comment:
`question`: can you `Deref` to `self.0`?

Reply:
Yes, but that would make `self.0` a public part of the API.

Plus it's typically not considered good style to use `Deref` for coercions when pointers aren't involved